Frontend/student-hub/src/pages/MentorshipPage.js

```diff
@@ -206,33 +230,33 @@ const delayTime = 60000;
   }
 
   const handlePendingRequest = () => {
-    const pendingRequest = mentorshipRequest.filter(
+    const pendingRequest = mentorshipRequests.filter(
       (request) => request.mentorship === 'Mentee' && request.status === 'requested'
     );
-    setFilteredMentorshipRequest(pendingRequest);
+    setFilteredMentorshipRequests(pendingRequest);
     setIsPendingRequestDialogOpen(true);
   }
 
   const handleOpenMenteeList = async () => {
-    setUserRole('Mentee')
+    setUserRole('Mentor')
```

Comment:
Why are we still setting userRole here? If we set it during the useEffect isn't this unneccesarily writing over that?